optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -170,7 +170,12 @@ def __init__(
         self._device = get_device_for_provider(encoder_session.get_providers()[0])
         self.encoder = ORTEncoder(session=encoder_session, device=self._device)
         self.decoder = ORTDecoder(session=decoder_session, device=self._device)
-        self.decoder_with_past = ORTDecoder(session=decoder_with_past_session, device=self._device)
+        self.use_past_key_values = decoder_with_past_session is not None
+        # If a decoder_with_past_path is provided, an inference session for the decoder with past key/values as inputs
+        # will be enabled
+        self.decoder_with_past = (
+            ORTDecoder(session=decoder_with_past_session, device=self._device) if self.use_past_key_values else None
+        )
         self.encoder_file_name = kwargs.get("last_encoder_model_name", ONNX_ENCODER_NAME)
         self.decoder_file_name = kwargs.get("last_decoder_model_name", ONNX_DECODER_NAME)
         self.decoder_file_with_past_name = kwargs.get("last_decoder_with_past_model_name", ONNX_DECODER_WITH_PAST_NAME)
```

Comment:
We should make this `None` when there is no past_key model file

Reply:
`self.use_past_key_values` is set to `True` when a `decoder_with_past_session` is given, `False` otherwise (it seems equivalent to what you are saying)